### PR TITLE
add lm_eval, enable pruning for CHN models

### PR DESCRIPTION
**Pull request by WeiweiZhang1**
## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/requirements.txt [hunk 1]
```diff
@@ -6,4 +6,9 @@ torch
 tqdm
 optimum
 einops
-
+intel_extension_for_transformers
+tiktoken
+transformers_stream_generator
+peft
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/requirements.txt:12]: Why do we require Perf and Deepspeed? Perhaps some other libraries aren't needed either

File: neural_compressor/compression/pruner/pruning.py [hunk 2]
```diff
@@ -228,12 +228,13 @@ def _do_pruning(self):
                     pruner.fasterprune(layer_op_names)
                 for j in range(len(inputs)):
                     # the weights of current layer have been pruned, get the latest outputs as the inputs for next layer
-                    input_infos = self.gather_single_batch_from_dict(inp_dict, j)
-                    inputs[j] = layer(inputs[j], **input_infos)[0]
+                    other_infos = self.gather_single_batch_from_dict(other_input_infos, j)
```
Comment (wenhuach21) [inline neural_compressor/compression/pruner/pruning.py:231]: move the following no grad to funciton, too much space make the code a little ugly
        inputs, positional_inputs, other_input_infos = collect_layer_inputs(
            model=self._model, layers=layers, layer_idx=0, layer_inputs=self._dataloader, device=self.dev
        )

        **with torch.no_grad():**
            for i in tqdm(range(len(layers))):

**Review by hshen14 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md [hunk 3]
```diff
@@ -68,29 +70,34 @@ The last word acc of the channel-wise sparse model is shown in the following tab
...
 | facebook/opt-6.7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.6769 | 0.6616 | -2.26% |
-| facebook/opt-6.7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.6730 | 0.6577 | -2.27% |
-| tiiuae/falcon-7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.7467 | 0.7528 | -0.82% |
-| tiiuae/falcon-7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.7464 | 0.7502 | -0.51% |
+| facebook/opt-6.7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.6730 | 0.6577 | -2.84% |
+| tiiuae/falcon-7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.7467 | 0.7528 | +0.82% |
+| tiiuae/falcon-7b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.7464 | 0.7502 | +0.47% |
 | bigscience/bloom-7b1 | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.5764 | 0.5606 | -2.74% |
-| bigscience/bloom-7b1 | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.5725 | 0.5587 | -2.41% |
+| bigscience/bloom-7b1 | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.5725 | 0.5587 | -3.07% |
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md:90]: have we tried with pile calibration?

**Review by wenhuach21 — APPROVED** [review]

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md [hunk 4]
```diff
@@ -68,29 +70,34 @@ The last word acc of the channel-wise sparse model is shown in the following tab
...
 | Model | Task | Calibration dataset | Evaluation dataset | Sparsity | Precision | Dense last word accuracy | Sparse last word accuracy | Relative drop |
 |  :----: | :----: | :----: | :----: | :----: | :----: | :----: |:----: |:----:|
+| meta-llama/Llama-2-7b-hf | CLM | wikitext-2-raw-v1 | lambada_openai | 30% | FP32 | 0.7392 | 0.7320 | -0.97% |
+| meta-llama/Llama-2-7b-hf | CLM | wikitext-2-raw-v1 | lambada_openai | 30% | BF16 | 0.7365 | 0.7304 | -1.19% |
 | EleutherAI/gpt-j-6b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.6831 | 0.6922 | +1.33% |
-| EleutherAI/gpt-j-6b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.6771 | 0.6874 | +1.52% |
+| EleutherAI/gpt-j-6b | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.6771 | 0.6874 | +0.63% |
 | decapoda-research/llama-7b-hf | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | FP32 | 0.7361 | 0.7332 | -0.39% |
-| decapoda-research/llama-7b-hf | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.7326 | 0.7297 | -0.23% |
+| decapoda-research/llama-7b-hf | CLM | wikitext-2-raw-v1 | lambada_openai | 40% | BF16 | 0.7326 | 0.7297 | -0.87% |
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md:84]: change to huggy/llama-7b later

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py [hunk 5]
```diff
@@ -621,7 +514,8 @@ def group_texts(examples):
         pruning_configs=[
             {
                 "pruning_type": "sparse_gpt",
-                "op_names": [".attn", "_proj", ".fc", "key", "dense", "_h"],
+                "op_names": [".*"],
+                "excluded_op_names": ["lm_head", "embed_out"],
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py:518]: is it ok to remove this? Sparsegpt should only handle layers in block

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py [hunk 6]
```diff
@@ -484,26 +379,24 @@ def main():
...
-    is_t5 = bool("t5" in args.model_name_or_path)
     if args.tokenizer_name:
         tokenizer = AutoTokenizer.from_pretrained(args.tokenizer_name, use_fast=not args.use_slow_tokenizer)
     elif args.model_name_or_path:
         if is_llama:
             tokenizer = transformers.LlamaTokenizer.from_pretrained(args.model_name_or_path)
         else :
-            tokenizer = AutoTokenizer.from_pretrained(args.model_name_or_path, use_fast=not args.use_slow_tokenizer, trust_remote_code=True)
+            tokenizer = AutoTokenizer.from_pretrained(args.model_name_or_path, 
+                                                      use_fast=not args.use_slow_tokenizer, trust_remote_code=True)
```
Comment (wenhuach21) [inline examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py:389]: When is it appropriate to use the fast tokenizer? Is it specific to the model being used, and can we provide additional comments to help the user?

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py (hunk 5 above)
Comment (WeiweiZhang1) [reply examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/run_clm_sparsegpt.py:518]: I've tested it, everything looks fine.

File: examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md (hunk 3 above)
Comment (WeiweiZhang1) [reply examples/pytorch/nlp/huggingface_models/language-modeling/pruning/eager/README.md:90]: Will do

**Pull request merged**